frontend/src/views/CellCards/components/CellCard/components/FullScreenProvider.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
```

Comment:
this custom provider yields a hook that can be used to manage full screen state in its child components.